platforms/ide/problems-api/src/main/java/org/gradle/api/problems/ProblemCategory.java

```diff
@@ -23,34 +23,55 @@
 
 /**
  * A category is a component of a {@link Problem} that helps group related problems together.
- *
- * The category string is a ":" separated string that starts either with a specific problem category or with "gradle-plugin"
- * and the plugin id where the problem is raised followed by the specific category.
- * <pre>{@code
- *  structure:
- *  <category>:<detailed-info>:<detailed-info>:...
- *  gradle-plugin:<plugin-id>:<category>:<detailed-info>:<detailed-info>:...
- * }</pre>
- *
- * <p>examples of valid category strings:
- * <pre>{@code
- * deprecation
- * gradle-plugin:deprecation
- * gradle-plugin:deprecation:<detailed-info>
- * gradle-plugin:deprecation:<detailed-info>:<detailed-info>:...
+ * <p>
+ * A category defines the following hierarchical elements to distinguish instances:
+ * <ul>
+ *     <li>namespace</li>
+ *     <li>category</li>
+ *     <li>subcategories</li>
+ * </ul>
+ * <p>
+ * The namespace denotes which component emitted the problem. It can be either Gradle core or a third-party plugin. The values should use the following pattern:
+ * <ul>
+ *     <li>Gradle core: {@code gradle}</li>
+ *     <li>Plugins: {@code gradle-plugin:<plugin-id>}</li>
+ * </ul>
+ * <p>
+ * A category should contain the most broad term describing the problem.
+ * A few examples are: {@code compilation}, {@code deprecation}, {@code task-validation}.
+ * <p>
+ * The problem category can be refined with an optional hierarchy of subcategories.
+ * For example, a problem covering a java compilation warning can be denoted with the following subcategories: {@code [java, unused-variable]}.
+ * <p>
+ * The categorization depends on the domain and don't have any constraints. Clients (i.e. IDEs) receiving problems should use the category information for
+ * properly group and sort the received instances.
  * }</pre>
  *
  * @since 8.5
  */
 @Incubating
 public interface ProblemCategory {
-    boolean hasPluginId();
-
-    String getPluginId();
 
+    /**
+     * The namespace.
+     *
+     * @return The namespace.
+     */
     String getNamespace();
 
+    /**
+     * The main category.
+     *
+     * @return The main categories.
+     */
     String getCategory();
 
+    /**
+     * The optional subcategories.
```

Comment:
🤔 Could these methods be combined with the understanding that the main category is element 0?  Seems like it would be annoying to make two calls, or to concatenate these to get the complete category path.

Reply:
On the IDE side, I think it's a nicer to read `problem.getCategory().getCategory()` rather than `problem.getCategory().getcategory.get(0)`. I'll soon play around with the integration, and I'll have a better notion on that.